src/Spice86.Core/Emulator/VM/Machine.cs

```diff
@@ -200,7 +205,9 @@ public sealed class Machine : IDisposable, IDebuggableComponent {
             Memory.UInt16[0xF000, 0xFFF0] = 0xF4;
         }
         IoPortDispatcher = ioPortDispatcher;
-        BiosDataArea = new BiosDataArea(Memory);
+        BiosDataArea = new BiosDataArea(Memory) {
+            MemSizeKb = (ushort)Math.Clamp(Memory.Ram.Size / 1024, 0, 1024) // max 1mb.
```

Comment:
Looks like this is not supposed to include ram above 640k
http://www.techhelpmanual.com/184-int_12h__conventional_memory_size.html
0x280

Reply:
Thanks, fixed. 
My source only said to use other interrupts for memory above 1mb. 640K makes more sense.